src/utilities/set-random-voice-channel-names.ts

```diff
@@ -0,0 +1,46 @@
+import { Client, Collection, StageChannel, VoiceChannel } from "discord.js";
+import { readFileSync } from 'fs';
+import * as path from 'path';
+
+export function setRandomVoiceChannelNames(client: Client): void {
+  // Parse & filter groups from file
+  const [miscNameGroup, ...nameGroups] = readFileSync(
+    path.join(__dirname, '../assets/voice-channel-names.txt')
+  )
+    .toString()
+    .split(/[\n]{2,}/gm)
+    .map(nameGroup => nameGroup.split('\n'))
+    .filter((nameGroup) => !nameGroup.every(s => s.startsWith('//')));
+  
+  // Get random group
+  const nameGroup = nameGroups[Math.floor(Math.random() * nameGroups.length)];
+
+  // Get voice channels for the current server
+  const voiceChannels = client.channels.cache.filter(c => c.isVoice()) as Collection<string, VoiceChannel | StageChannel>;
+  const numVoiceChannels = voiceChannels.size;
+
+  // Get list of random names
+  const randomNames: string[] = [];
+  for (let i = 0; i < numVoiceChannels; i++) {
+    let group
+    if (nameGroup.length) {
+      group = nameGroup;
+    } else if (miscNameGroup.length) {
+      group = miscNameGroup;
```

Comment:
Does this mean that if we don't have enough groups, we pick a random egg channel? It seems like it might be weird if only one or two channels need to be egg channels and they end up being Egg 6 and Egg 1 for example.